src/dappsExplorer/DAppsExplorerScreenSearchFilter.test.tsx

```diff
@@ -39,670 +40,680 @@ const defaultStore = createMockStore({
-    expect(queryByText('featuredDapp')).toBeFalsy()
-    expect(getByText('Dapp 1')).toBeTruthy()
-    expect(getByText('Dapp 2')).toBeTruthy()
-
-    fireEvent.press(getByText('Dapp 1'))
-
-    expect(defaultStore.getActions()).toEqual([
-      fetchDappsList(),
-      dappSelected({ dapp: { ...dappsList[0], openedFrom: DappSection.All } }),
-    ])
-  })
-
-  it('renders correctly and fires the correct actions on press deep linked dapp', () => {
-    const { getByText, queryByText } = render(
-      <Provider store={defaultStore}>
-        <DAppsExplorerScreenSearchFilter />
-      </Provider>
-    )
-
-    expect(defaultStore.getActions()).toEqual([fetchDappsList()])
-    expect(getByText('Dapp 1')).toBeTruthy()
-    expect(getByText('Dapp 2')).toBeTruthy()
-    expect(queryByText('featuredDapp')).toBeFalsy()
-
-    fireEvent.press(getByText('Dapp 2'))
-
-    expect(defaultStore.getActions()).toEqual([
-      fetchDappsList(),
-      dappSelected({ dapp: { ...dappsList[1], openedFrom: DappSection.All } }),
-    ])
-  })
-
-  it('pens dapps directly', () => {
-    const store = createMockStore({
-      dapps: {
-        dappListApiUrl: 'http://url.com',
-        dappsList,
-        dappsCategories,
-      },
+describe.each([{ isTabNavigator: false }, { isTabNavigator: true }, {}])(
+  'DAppsExplorerScreenSearchFilter (isTabNavigator: $isTabNavigator)',
+  (routeParams) => {
+    beforeEach(() => {
+      defaultStore.clearActions()
+      jest.clearAllMocks()
     })
-    const { getByText } = render(
-      <Provider store={store}>
-        <DAppsExplorerScreenSearchFilter />
-      </Provider>
-    )
 
-    expect(getByText('dappsDisclaimerAllDapps')).toBeTruthy()
-
-    fireEvent.press(getByText('Dapp 1'))
-
-    expect(store.getActions()).toEqual([
-      fetchDappsList(),
-      dappSelected({ dapp: { ...dappsList[0], openedFrom: DappSection.All } }),
-    ])
-  })
-
-  describe('favorite dapps', () => {
-    it('renders correctly when there are no favorite dapps', () => {
-      const store = createMockStore({
-        dapps: {
-          dappListApiUrl: 'http://url.com',
-          dappsList,
-          dappsCategories,
-          favoriteDappIds: [],
-        },
-      })
-      const { getByText, queryByText } = render(
-        <Provider store={store}>
-          <DAppsExplorerScreenSearchFilter />
+    it('renders correctly and fires the correct actions on press dapp', () => {
+      const { getByText, queryByText, getByTestId, queryByTestId } = render(
+        <Provider store={defaultStore}>
+          <MockedNavigator component={DAppsExplorerScreenSearchFilter} params={routeParams} />
         </Provider>
       )
 
-      expect(queryByText('dappsScreen.noFavorites.title')).toBeNull()
-      expect(queryByText('dappsScreen.noFavorites.description')).toBeNull()
+      // TODO (ACT-1108): should be true in all cases
+      // Shows the QR button
+      if (!routeParams.isTabNavigator) {
```

Comment:
This is a bit hacky, but will be true in all cases once act-1108 is created and can be removed then.